fission/src/ui/components/SceneOverlayEvents.ts

```diff
@@ -0,0 +1,61 @@
+let nextTagId = 0
+
+/* Coordinates for tags in world space */
+export type PixelSpaceCoord = [number, number]
+
+/**
+ * Represents a tag that can be displayed on the screen
+ *
+ * @param text The text to display
+ * @param position The position of the tag in screen space (default: [0,0])
+ */
+export class SceneOverlayTag {
+    private _id: number
+    public text: string
+    public position: PixelSpaceCoord // Screen Space
+
+    public get id() {
+        return this._id
+    }
+
+    /** Create a new tag */
+    public constructor(text: string, position?: PixelSpaceCoord) {
+        this._id = nextTagId++
+
+        this.text = text
+        this.position = position ?? [0, 0]
+    }
+
+    /** Update the tag's text */
+    public Update() {
+        new SceneOverlayTagEvent(this)
+    }
```

Comment:
This updates position too, yeah? With this being called every frame with a robot in it, and there being only one listener, could we remove the event middleman and call the update directly?

Reply:
The update function allows the SceneOverlay to get the new updated tag so I can't call it from SceneOverlay directly

Reply:
All of the tags get updated though, and this update does the entire map for each tag. If MirabufSceneObject adjusted a public map instead of sending the update event, SceneOverlay could just use the map and do one update for all of them at once.